Add de.thejackimonster.KnockKnock

TheJackiMonster opened the pull request:
<!-- ⚠️⚠️  Submission pull request MUST be made against the `new-pr` **base branch** ⚠️⚠️  -->

<!-- 💡 Please go to the preview tab to view the markdown below 💡 -->

### Please confirm your submission meets all the criteria

<!-- 💡 Please replace each `[ ]` with `[X]` when the step is complete 💡 -->

<!-- 💡 Please tick and write 'N/A' with a reason if a checklist item below is not applicable 💡 -->

- [x] Please describe the application briefly. *-> The application is intended to control doors or other simple devices via SSH trigger commands via graphical interface.*
- [x] Please attach a video showcasing the application on Linux using the Flatpak. https://github.com/user-attachments/assets/d0740d86-0b10-41eb-9fb3-8bbfaed74351
- [x] The Flatpak ID follows all the rules listed in the [Application ID requirements][appid].
- [x] I have read and followed all the [Submission requirements][reqs] and the [Submission guide][reqs2] and I agree to them.
- [x] I am an _(please keep whichever is applicable and remove the rest)_ author/developer/upstream contributor to the project.

<!-- 💡 Please mention below the GitHub usernames of any additional maintainers needed (if any) 💡 -->

<!-- ⚠️⚠️  Please DO NOT modify anything below this line ⚠️⚠️  -->

[appid]: https://docs.flathub.org/docs/for-app-authors/requirements#application-id
[reqs]: https://docs.flathub.org/docs/for-app-authors/requirements
[reqs2]: https://docs.flathub.org/docs/for-app-authors/submission

hfiguiere commented:
>  Please attach a video showcasing the application on Linux using the Flatpak. -> N/A, the application is only usable inside a network with a fitting door/device.

Your application is not special.

TheJackiMonster commented:
> > Please attach a video showcasing the application on Linux using the Flatpak. -> N/A, the application is only usable inside a network with a fitting door/device.
> 
> Your application is not special.

So I can either record a video showcasing the application running without a proper device in local network. That will make it only behave as dummy for the most part... or I record a video with some device in local network and it may contain details which should not get published (like IP address or identifiers for SSH connections), so I would likely need to censor them in the video.

TheJackiMonster commented on de.thejackimonster.KnockKnock.json:
```diff
@@ -0,0 +1,38 @@
+		"--share=network",
+		"--socket=ssh-auth",
+		"--filesystem=host-os",
+		"--env=PATH=/app/bin:/usr/bin:/run/host/usr/bin:/run/host/bin"
```
Okay, so I've added iputils as module which works as well.

TheJackiMonster commented:
I've attached a video now, I added a module so that the flatpak does not require access to the ping binary from the host and I added instructions to properly install the license files.

bbhtt commented on de.thejackimonster.KnockKnock.json:
```diff
@@ -0,0 +1,59 @@
+	"command": "knock-knock",
+	"finish-args": [
+		"--socket=wayland",
+		"--socket=fallback-x11",
```
```suggestion
		"--socket=fallback-x11",
		"--share=ipc",
```

bbhtt commented on de.thejackimonster.KnockKnock.json:
```diff
@@ -0,0 +1,59 @@
+{
+	"app-id": "de.thejackimonster.KnockKnock",
+	"runtime": "org.gnome.Platform",
```
why does it need gnome runtime?

bbhtt replied:
GTK3 or 4? GTK 3 is in freedesktop sdk and libhandy is easy to build.

TheJackiMonster replied:
But doesn't the GNOME runtime already ship libhandy and GTK3? Is it really better to use freedesktop and add additional dependencies manually to maintain than relying on generally available runtimes that would increase overall redundant usage of them decreasing additional space required for flatpaks on the host?

I pretty much see arguments for both ways here.

TheJackiMonster commented on de.thejackimonster.KnockKnock.json:
```diff
@@ -0,0 +1,59 @@
+			],
+			"sources": [
+				{
+					"type": "archive",
+					"url": "https://gitlab.com/TheJackiMonster/knock-knock/-/archive/v1.0.2/knock-knock-v1.0.2.tar.gz",
+					"sha512": "40ec6a0b0ddc5536971a86e3d239be33b5613eabe36375704835a120d5f3e949f3172de05a1dcefafdb1284fc276c1955a6af2d33ad9767f47f5b208304e4722",
```
Yes, pretty much.

TheJackiMonster replied:
But in that case it doesn't make much sense, does it? Only because I don't split multiple classes into different source code files, doesn't mean it has reduced complexity. It's no fork and I also created appdata files as well as icon. Even the UI is separated in files from Glade.

So why would this apply here?
